src/battle_script_commands.c

```diff
@@ -16748,12 +16748,22 @@ void BS_TryRelicSong(void)
     }
 }
 
+static const struct PledgeCombo sPledgeCombos[] =
+{
+    // Dominant,  Passive,    Script to execute
+    { TYPE_GRASS, TYPE_WATER, BattleScript_EffectCombinedPledge_Grass },
+    { TYPE_WATER, TYPE_FIRE,  BattleScript_EffectCombinedPledge_Water },
+    { TYPE_FIRE,  TYPE_GRASS, BattleScript_EffectCombinedPledge_Fire },
```

Comment:
Do you think you could find a way to include base power in the table as well in this table and then store it in a BattleStruct field?
```
    case EFFECT_PLEDGE:
        if (gBattleStruct->pledgeMove)
            basePower = BattleStruct->combinedMoveBasePower;
```
1. It can change be adjusted depending on the type
2. If hackers add more type combinations they aren't locked into 150

Reply:
Great idea overall

Reply:
> Do you think you could find a way to include base power in the table as well in this table and then store it in a BattleStruct field?

I was able to do this locally, but I want to check what else would we like to allow for customability.
Targets? Animations? This could potentially be a realization of GF's old "combined moves" leftover code.

Reply:
I doubt target is needed. That's get chosen by AI / Player. 

Not sure about animation since usually the one that overrules the type get chosen.

Reply:
I would say that the target could change, in the form of going from  `MOVE_TARGET_SELECTED` to `MOVE_TARGET_BOTH` or `MOVE_TARGET_FOES_AND_ALLY` (or even `MOVE_TARGET_SELF`).
Testing by replacing the effect for Fire+Grass with `BattleScript_EffectExplosion` only damaged 1 opponent. And used the animation for Fire Pledge.

Reply:
I've been ruminating on this for a bit and I wonder what would be the best solution, as calling `BattleScript_EffectExplosion` would technically be calling the "X used Y!" text again as part of its own script, so potential foot guns might been set up.

Reply:
It could, ye. The effect has to go through the same checks `BattleScript_EffectPledge` does including `attackcancler` which might lead to problems but if you want to avoid that you can't allow unique scripts in which case I think it should be users responsibility because we should give the the freedom to make new `Pledge` effects. We could leave a comment to to guide them writing similar scripts as `EffectCombinedPledge_X`

Generally, I had a thought that it might make more sense to use move IDs instead of Types (personally I wouldn't consider it hard coding in this case because you want an actual move result). With types you are still bounded to those limitations and move ids give you flexibility and allow unique combinations

To take an example from Lunos repo. Ember + Gust could be turned into Heat Wave and you would avoid the issue of turning Ember + Water Pledge into Water Pledge. 

This is just an idea (not mine), so not blocking.